### DISCUSSION TITLE
Rendering a custom format inside a project.

### DISCUSSION THREAD

**Discussion by andrewpbray** [quarto-dev/quarto-cli#6837] [Q&A] [answered]
### Description

I have a project with an `_extensions` dir at the root level that contains several custom formats. I have a qmd file that uses one of those custom formats buried in a subdir. I've noticed different behavior when I render it in different ways:

1. If I run `quarto render path/to/file.qmd` it finds the extension, renders just fine, and puts the output in `_site`.
2. If I run `quarto preview`, it finds the extension and renders just fine.
3. If I click the Render arrow in RStudio on `file.qmd` it errors out, not being able to find the extension file (it looked it the same dir as `file.qmd`.

What exactly is the quarto command that's being run by clicking the Render arrow that it isn't project/extension - aware?

**Comment by mcanouil** [comment]
Are you using a Quarto project (presence of `_quarto.yml`)?

Could you share a Git repository with your setup?

**Reply by andrewpbray** [reply]
Yep, this is a website project. Here's an example of the file (and repo) described above:

[https://github.com/posit-conf-2023/quarto-r/blob/main/materials/1-single-docs/1-welcome-to-quarto/slides.qmd](https://github.com/posit-conf-2023/quarto-r/blob/main/materials/1-single-docs/1-welcome-to-quarto/slides.qmd)

**Comment by cderv** [comment, marked as answer]
> If I click the Render arrow in RStudio on file.qmd it errors out, not being able to find the extension file (it looked it the same dir as file.qmd.

it seems this is an issue with RStudio IDE itself. It would probably be worth opening in RStudio IDE repo: https://github.com/rstudio/rstudio
Team there would know the best what is exactly different in the way the render button works.
